apps/hash-frontend/src/pages/new/types/entity-type.page.tsx

```diff
@@ -82,6 +99,13 @@ const Page: NextPageWithLayout = () => {
     },
   });
 
+  const [initialPropertyTypes, setInitialPropertyTypes] = useState<
+    (PropertyTypeDefinition | PropertyType)[]
+  >([]);
+
+  const entityTypeTitle = watch("name");
+  const entityTypeDescription = watch("description");
```

Comment:
If possible, it would be better to use `useWatch` for these – inside the `SelectGeneratedPropertyTypes` component – for two reasons:

1. Whenever you use the `watch` method returned by `useForm/useFormContext` – it subscribes at the top level of the form. The entire component in which `useForm` is called re-renders whenever the value you've watched changes – not the component from which you've called it in. In this case, they're one and the same (though point 2 explains why it shouldn't be), but it means using `watch` itself instead of `useWatch` is a bit of a red flag
2. These values aren't used in this component directly – they're only used by the child component. The philosophy of react-hook-form is that you minimise component re-renders as a user interacts with a form by using uncontrolled inputs where-ever possible, and subscribing to only the bits of the form you need at the lowest possible level in your component tree. 

In this particular form, it probably doesn't matter as performance is likely to be fine no matter what you do. But these are useful principles to stick to as it what enables react hook form to perform fairly well in more complex cases, so it's a good idea to follow the principles universally to get used to them, so you don't end up with poor performance by accident with more complex forms.

Reply:
This behaviour is described at https://react-hook-form.com/api/useform/watch/

> This API will trigger re-render at the root of your app or form, consider using a callback or the [useWatch](https://react-hook-form.com/api/usewatch/) api if you are experiencing performance issues.

Reply:
Thanks for the suggestions, not super familiar with `react-form-hook` yet. I've refactored the code to make use of `useWatch` in the `SelectGeneratedPropertyTypes` component in https://github.com/hashintel/hash/pull/2486/commits/1f948bee67b202b118845ca39c46f32e386e3589. Slightly improved the types in https://github.com/hashintel/hash/pull/2486/commits/866d9c361882bfa6fde601ff60f51e9ed5d63a27